src/SFA.DAS.AssessorService.Application/Handlers/Certificates/GetCertificatesHistoryHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SFA.AssessorService.Paging;
+using SFA.DAS.AssessorService.Api.Types.Models.Certificates;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Entities;
+using SFA.DAS.AssessorService.Domain.JsonData;
+using SFA.DAS.AssessorService.ExternalApis;
+using SFA.DAS.AssessorService.ExternalApis.AssessmentOrgs;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Certificates
+{
+    public class GetCertificatesHistoryHandler : IRequestHandler<GetCertificateHistoryRequest, PaginatedList<CertificateHistoryResponse>>
+    {
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly IContactQueryRepository _contactQueryRepository;
+        private readonly IAssessmentOrgsApiClient _assessmentOrgsApiClient;
+        private readonly ILogger<GetCertificatesHistoryHandler> _logger;
+
+        public GetCertificatesHistoryHandler(ICertificateRepository certificateRepository,
+            IContactQueryRepository contactQueryRepository,
+            IAssessmentOrgsApiClient assessmentOrgsApiClient,
+            ILogger<GetCertificatesHistoryHandler> logger)
+        {
+            _certificateRepository = certificateRepository;
+            _contactQueryRepository = contactQueryRepository;
+            _assessmentOrgsApiClient = assessmentOrgsApiClient;
+            _logger = logger;
+        }
+
+        public async Task<PaginatedList<CertificateHistoryResponse>> Handle(GetCertificateHistoryRequest request, CancellationToken cancellationToken)
+        {
+            const int pageSize = 10;
+
+            var contact = await _contactQueryRepository.GetContact(request.Username);
+
+            var certificates = await _certificateRepository.GetCertificateHistory(
+                contact.OrganisationId.Value,
+                request.PageIndex ?? 1,
+                pageSize);
```

Comment:
Can we get rid of the extra db call to GetContact and just pass the username straight through to GetCertHistory and then join on Contacts?  Quite happy for GetCertificateHistory to be a nice Dapper SQL statement if that's easier than trying to bash EF into what you need to do. 👍

Reply:
is ok I can just join is easy - will do

Reply:
done